File: STRUKTUR DATA (SMSTR 2)/materi UAS/queue_linkedlistbased.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

class Queue:
    def __init__(self):
        self.front = None
        self.rear = None

    def enqueue(self):
        val = int(input("Menambahkan data kedalam queue: "))
        new_node = Node(val)
        if self.rear is None:
            self.front = self.rear = new_node
        else:
            self.rear.next = new_node
            self.rear = new_node

    def dequeue(self):
        if self.front is None:
            print("Antrian kosong, tidak bisa dequeue")
            return
        else:
            print("Data yang keluar:", self.front.data)
            self.front = self.front.next
            if self.front is None:
                self.rear = None

    def display(self):
        if self.front is None and self.rear is None:
            print("Antrian kosong.")
            return
        print("Data dalam antrian adalah:", end=" ")
        temp = self.front
        while temp is not None:
            print(temp.data, end=" ")
            temp = temp.next
        print()

def main():
    queue = Queue()
    done = False
    while not done:
        print("\n1) Enqueue")
        print("2) Dequeue")
        print("3) Tampilkan data antrian")
        print("4) Selesai")
        choice = int(input("Pilih: "))
        if choice == 1:
            queue.enqueue()
        elif choice == 2:
            queue.dequeue()
        elif choice == 3:
            queue.display()
        elif choice == 4:
            print("Keluar")
            done = True
        else:
            print("Coba lagi ya!!!")

if __name__ == "__main__":
    main()

File: STRUKTUR DATA (SMSTR 2)/materi UAS/queue_linkedlistbased.cpp
//QUEUE LINKED LIST BASED
//MENU

#include<iostream>
#include<stdlib.h>
#include<conio.h> //kadang kadang bisa digunakan kadang2 gabisa dgunakan. kalo conio dipake, pake getch juga

using namespace std;
struct node {
    int data;
    struct node *next;
};
struct node* front = NULL;
struct node* rear = NULL;
struct node* temp;

void Insert() {
    int val;
    cout<<"Menambahkan data kedalam queue : "<<endl;
    cin>>val;
    if (rear == NULL) {
        rear = (struct node *)malloc(sizeof(struct node));
        rear -> next = NULL;
        rear -> data = val;
        front = rear;
    } else {
        temp = (struct node *)malloc(sizeof(struct node));
        rear -> next = temp;
        temp -> data = val;
        temp -> next = NULL;
        rear = temp;
    }
}
void Delete() {
    temp = front;
    if (front == NULL) {
        cout<<"Antrian kosong, tidak bisa dequeue"<<endl;
        getch();
        return;
    } else
    if (temp -> next != NULL) {
        temp = temp -> next;
        cout<<"Data yang keluar: "<<front -> data<<endl;
        free(front);
        front = temp;
    } else {
        cout<<"Data yang keluar adalah: "<<front -> data<<endl;
        free(front);
        front = NULL;
        rear = NULL;
    }
    getch();
}
void Display() {
    temp = front;
    if ((front == NULL) && (rear == NULL)) {
        cout<<"Antrian kosong."<<endl;
        getch();
        return;
    }
    cout<<"Data dalam antrian adalah: ";
    while (temp != NULL){
        cout<<temp -> data<<" ";
        temp = temp -> next;
    }
    cout<<endl;
    getch();
}

int main(){
    int ch;
    do {
        system("CLS");
        cout<<"1) Enqueue"<<endl;
        cout<<"2) Dequeue"<<endl;
        cout<<"3) Tampilkan data antrian"<<endl;
        cout<<"4) Selesai"<<endl;
        cout<<"pilih: "<<endl;
        cin>>ch;
        switch (ch) {
            case 1: Insert();
            break;
            case 2: Delete();
            break;
            case 3: Display();
            break;
            case 4: cout<<"Keluar"<<endl;
            break;
            default: cout<<"Coba lagi ya!!!"<<endl;
        }
    } while(ch != 4);
    return 0;
}